### DISCUSSION TITLE
Fails installing laws with kong 2.7 as the base image

### DISCUSSION THREAD

**Discussion by kollol-chowdhury-incontact** [Kong/kong#8244] [Help] [answered]
Whenever I am trying to build a docker image with kong 2.7 as the base image and then try to install some lua libraries like laws,it gives error.

Installing https://luarocks.org/laws-0.1-0.all.rock
#15 6.291
#15 6.291 Error: Failed installing dependency: https://luarocks.org/luaossl-20200709-0.src.rock - Could not find header file for CRYPTO
#15 6.291   No file openssl/crypto.h in /usr/local/include
#15 6.291   No file openssl/crypto.h in /usr/include
#15 6.291   No file openssl/crypto.h in /include
#15 6.291 You may have to install CRYPTO in your system and/or pass CRYPTO_DIR or CRYPTO_INCDIR to the luarocks command.
#15 6.291 Example: luarocks install luaossl CRYPTO_DIR=/usr/local
#15 6.300 Missing dependencies for laws 0.1-0:
#15 6.300    luaossl (not installed)
#15 6.300
#15 6.300 laws 0.1-0 depends on lua (5.1-1 provided by VM)
#15 6.300 laws 0.1-0 depends on luaossl (not installed)
#15 6.300 Installing https://luarocks.org/luaossl-20200709-0.src.rock
------
executor failed running [/bin/sh -c luarocks install laws 0.1-0]: exit code: 1

I could do it with 2.0.3 but when i am trying to upgrade the kong version,i get this error.

Please help

**Comment by mayocream** [comment, marked as answer]
Hi @kollol-chowdhury-incontact, I am using the `kong:2.7.0` image, and these steps below might help you:

```bash
apk add curl git gcc musl-dev
luarocks install luaossl OPENSSL_DIR=/usr/local/kong CRYPTO_DIR=/usr/local/kong
luarocks install laws
```

In the latest release, we took resty-openssl in place of luaossl, see:
- https://github.com/Kong/kong/pull/5387

**Reply by kollol-chowdhury-incontact** [reply]
Thanks..It worked 👍
